### PR TITLE
Fix error propagation in checkandAddSnapshotFinalizers and add tests

### LINKED ISSUE #196: Improve code coverage for pkg/common-controller
Need to improve code coverage after the split controller refactor.
/assign @ggriffiths 
/cc @xing-yang 
/kind cleanup

### PR DESCRIPTION
**Pull request by ggriffiths**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Adds more tests, and fixes a few bugs I found.

**Which issue(s) this PR fixes**:
Fixes #196 

**Special notes for your reviewer**:
* I uncommented all commented tests, fixed them, and added additional ones where needed.
* code cov for snapshot_controller.go is now 82.9%, up from 74% previously.
* broke up the tests & bug fixes into separate commits

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/common-controller/snapshot_controller.go [hunk 1]
```diff
@@ -665,26 +671,6 @@ func (ctrl *csiSnapshotCommonController) updateSnapshotErrorStatusWithEvent(snap
 	return nil
 }
 
-// isSnapshotConentBeingUsed checks if snapshot content is bound to snapshot.
```
Comment (ggriffiths) [inline pkg/common-controller/snapshot_controller.go]: this isn't used anywhere

File: pkg/common-controller/snapshot_controller.go [hunk 2]
```diff
@@ -173,7 +173,11 @@ func (ctrl *csiSnapshotCommonController) syncSnapshot(snapshot *crdv1.VolumeSnap
 		}
 	} else {
 		klog.V(5).Infof("syncSnapshot[%s]: check if we should add finalizers on snapshot", utils.SnapshotKey(snapshot))
-		ctrl.checkandAddSnapshotFinalizers(snapshot)
+		if err := ctrl.checkandAddSnapshotFinalizers(snapshot); err != nil {
+			klog.Errorf("error check and add Snapshot finalizers for snapshot [%s]: %v", snapshot.Name, errFinalizer)
+			ctrl.eventRecorder.Event(snapshot, v1.EventTypeWarning, "SnapshotCheckandUpdateFailed", fmt.Sprintf("Failed to check and update snapshot: %s", err.Error()))
```
Comment (ggriffiths) [inline pkg/common-controller/snapshot_controller.go]: I used this warning name from a test that used to test this behavior (before the controller split).

File: pkg/common-controller/snapshot_delete_test.go [hunk 3]
```diff
@@ -130,159 +130,44 @@ var snapshotClasses = []*crdv1.VolumeSnapshotClass{
 func TestDeleteSync(t *testing.T) {
 	tests := []controllerTest{
 		{
-			name:              "1-1 - content with empty snapshot class is deleted if it is bound to a non-exist snapshot and also has a snapshot uid specified",
```
Comment (ggriffiths) [inline pkg/common-controller/snapshot_delete_test.go]: Many of these tests are no longer required in this package, or are duplicates of the new ones I wrote below (3-1, 3-2, etc)

File: pkg/common-controller/snapshot_controller.go [hunk 4]
```diff
@@ -301,7 +305,9 @@ func (ctrl *csiSnapshotCommonController) checkandAddSnapshotFinalizers(snapshot
 	if addSourceFinalizer || addBoundFinalizer {
 		// Snapshot is not being deleted -> it should have the finalizer.
 		klog.V(5).Infof("checkandAddSnapshotFinalizers: Add Finalizer for VolumeSnapshot[%s]", utils.SnapshotKey(snapshot))
-		ctrl.addSnapshotFinalizer(snapshot, addSourceFinalizer, addBoundFinalizer)
+		if err := ctrl.addSnapshotFinalizer(snapshot, addSourceFinalizer, addBoundFinalizer); err != nil {
```
Comment (yuxiangqian) [inline pkg/common-controller/snapshot_controller.go]: nit: how about simply? "return ctrl.addSnapshot....."

File: pkg/common-controller/snapshot_controller.go (hunk 1 above)
Comment (yuxiangqian) [reply pkg/common-controller/snapshot_controller.go]: +1

File: pkg/common-controller/framework_test.go [hunk 5]
```diff
@@ -1494,3 +1501,5 @@ func emptyDataSecretAnnotations() map[string]string {
 		utils.AnnDeletionSecretRefNamespace: "default",
 	}
 }
+
+func toStrPointer(str string) *string { return &str }
```
Comment (yuxiangqian) [inline pkg/common-controller/framework_test.go]: where is this function used? seems not in this PR

Comment (ggriffiths) [reply pkg/common-controller/framework_test.go]: Ah yeah, I was using it for a test originally but removed it's usage. Will remove this

**Comment by ggriffiths** [comment]
Addressed your suggestions @yuxiangqian. Thanks!

**Comment by yuxiangqian** [comment]
@ggriffiths please squash the commits. thanks

**Pull request merged**